File: 2023/01.lua
function partOne (lines) 
    sum = 0
    for _,line in pairs(lines) do
        line =  string.gsub(line, "%a", "")
        num = string.sub(line, 1, 1) .. string.sub(line, -1)
        sum = sum + num
    end
    return sum
end

function partTwo (lines) 
    return 'todo'
end

function main() 
    lines = {}
    for line in io.lines() do
        table.insert(lines, line)
    end
    print('partOne', partOne(lines))
    print('partTwo', partTwo(lines))
end
main()

File: 2025/01.py
"""2025 Day 1: Secret Entrance"""
import fileinput
import heapq
import re
from collections import deque, defaultdict, Counter
from functools import lru_cache
from utils import perf_timer

def part_one(instructions: list[tuple[str, int]]):
    """Solution to part one."""
    pos = 50
    zeroes = 0
    for d, n in instructions:
        if d == 'L':
            pos = (pos - n) % 100
        elif d == 'R':
            pos = (pos + n) % 100

        if pos == 0:
            zeroes += 1

    return zeroes



def part_two(lines):
    """Solution to part two."""
    return 'todo'


re_dir = re.compile(r'([LR])(\d+)')
def main():
    """Parse input file, pass to puzzle solvers."""
    instructions = []
    for line in fileinput.input():
        line = line.strip()
        if m := re_dir.match(line):
            d, n = m.groups()
            instructions.append((d, int(n)))

    print('part_one', part_one(instructions))

    print('part_two', part_two(instructions))


if __name__ == '__main__':
    main()
